File: codegen/results/v001/qwen2.5-coder_7b/20260409_103237/python/008_prime_numbers/solution.py
import sys

def is_prime(num):
    if num <= 1:
        return False
    if num <= 3:
        return True
    if num % 2 == 0 or num % 3 == 0:
        return False
    i = 5
    while i * i <= num:
        if num % i == 0 or num % (i + 2) == 0:
            return False
        i += 6
    return True

def print_primes_up_to_n(n):
    for number in range(2, n + 1):
        if is_prime(number):
            print(number)

if __name__ == "__main__":
    if len(sys.argv) != 2 or not sys.argv[1].isdigit():
        print("Usage: python script.py <integer>")
        sys.exit(1)
    
    N = int(sys.argv[1])
    print_primes_up_to_n(N)

File: codegen/results/v001/qwen2.5-coder_7b/20260409_103237/typescript/008_prime_numbers/solution.ts
function isPrime(num: number): boolean {
    if (num <= 1) return false;
    if (num <= 3) return true;

    if (num % 2 === 0 || num % 3 === 0) return false;

    for (let i = 5; i * i <= num; i += 6) {
        if (num % i === 0 || num % (i + 2) === 0) return false;
    }

    return true;
}

const N = parseInt(process.argv[2], 10);

if (!isNaN(N)) {
    for (let number = 2; number <= N; number++) {
        if (isPrime(number)) {
            console.log(number);
        }
    }
}